index.bs

```diff
@@ -331,7 +359,92 @@ The {{PerformanceLongAnimationFrameTiming/blockingDuration}} attribute's getter
     1. If |workDuration| + |renderDuration| is greater than 50, then return |workDuration| + |renderDuration| - 50 milliseconds.
     1. Return 0.
 
-</div>
+The {{PerformanceLongAnimationFrameTiming/scripts}} attribute's getter steps are:
+    1. Let |scripts| be « ».
+    1. [=list/For each=] |scriptInfo| in [=this=]'s [=frame timing info=]'s [=frame timing info/scripts=]:
+        1. Let |scriptEntry| be a new {{PerformanceScriptTiming}} in [=this=]'s [=relevant realm=],
+             whose [=PerformanceScriptTiming/timing info=] is |scriptInfo|.
+        1. [=list/Append=] |scriptEntry| to |scripts|.
+    1. Return |scripts|.
+
+{{PerformanceScriptTiming}} interface {#sec-PerformanceScriptTiming}
+------------------------------------------------------------------------
+
+<pre class="idl">
+    enum ScriptTimingType {
+        "classic-script",
+        "module-script",
+        "event-listener",
+        "user-callback",
+        "resolve-promise",
+        "reject-promise"
+    };
+
+    [Exposed=Window]
+    interface PerformanceScriptTiming : PerformanceEntry {
+        /* Overloading PerformanceEntry */
+        readonly attribute DOMHighResTimeStamp startTime;
+        readonly attribute DOMHighResTimeStamp duration;
+        readonly attribute DOMString name;
+        readonly attribute DOMString entryType;
+
+        readonly attribute ScriptTimingType type;
+        readonly attribute DOMHighResTimeStamp executionStart;
+        readonly attribute DOMString sourceLocation;
+        [Default] object toJSON();
+    };
+</pre>
+
+A {{PerformanceScriptTiming}} has an associated [=script timing info=] <dfn for=PerformanceScriptTiming>timing info</dfn>.
+
+The {{PerformanceScriptTiming/entryType}} attribute's getter step is to return <code>"script"</code>.
+
+The {{PerformanceScriptTiming/type}} attribute's getter step is to return [=this=]'s [=PerformanceScriptTiming/timing info=]'s [=script timing info/type=].
+
+The {{PerformanceScriptTiming/name}} attribute's getter steps are:
+    1.  Switch on |this|'s {{PerformanceScriptTiming/type}}:
+
+        : "`classic-script`"
+        : "`module-script`"
+        :: Return |this|'s [=PerformanceScriptTiming/timing info=]'s [=script timing info/source url=].
+
+        : "`event-listener`"
+        ::
+            1. Let |targetName| be |this|'s [=PerformanceScriptTiming/timing info=]'s [=script timing info/invoker name=].
```

Comment:
Is this pattern used elsewhere to generate selectors to identify elements?

Reply:
Not sure. We can discuss whether it's the right choice or we can do something else.

Reply:
Added an inline issue for this, for now I want the spec to reflect what's implemented, and then we can discuss and change. Does that sound OK?